src/com/jwetherell/algorithms/strings/StringFunctions.java

```diff
@@ -21,6 +26,9 @@ public static final String reverseWithStringConcat(String string) {
     }
 
     public static final String reverseWithStringBuilder(String string) {
+        if (string == null || string.isEmpty()){
```

Comment:
Added check to return empty string or NULL to prevent exceptions during code runtime.